### PR TITLE
1008 Add formulas NB.NB.2, 7, 8, 9, 11 and 12 from Eurocode 1993-1-1 NB NL

### LINKED ISSUE #1008: [🔧 Formula Request]: Add formulas NB.NB.2, 7, 8, 9, 11 and 12 from Eurocode 1993-1-1 NB NL.
### Formula Description

Add formulas NB.NB.2, 7, 8 and 9 from Eurocode 1993-1-1 NB NL.

### Acceptance Criteria

- [x] Add the formula to the code
- [x] Add latex expression to the formula
- [x] Add unittests
- [x] Add docstrings
- [x] Update table in docs with classname(s)

### Dependencies

_No response_

### Confirmation

- [x] I have checked that this formula does not already exist in the issue tracker.
- [x] I have read and understood the contribution guidelines.

### PR DESCRIPTION
**Pull request by GerjanDorgelo**
## Description

Add formulas NB.NB.2, 7, 8, 9, 11 and 12 from Eurocode 1993-1-1 NB NL

Fixes #1008

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## Checklist:

- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] New and existing unit tests pass locally with my changes

## Release Notes

* **New Features**
  * Added six Dutch National Annex formulas for steel beam buckling design, covering critical buckling moment, reduction factors, parameter S, coefficient C, and alpha.
  * Added support for sixth-power length units in engineering calculations.

* **Documentation**
  * Updated Eurocode formula documentation to show these calculations as available.

* **Tests**
  * Added coverage for numerical calculations, input validation, boundary conditions, and formula rendering.

### CODE REVIEW

File: blueprints/codes/eurocode/national_annex/nl/nen_en_1993_1_1_2006/chapter_NB_NB_critical_buckling_moment/formula_nb_nb_11.py [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+    def _evaluate(
+        c_1: DIMENSIONLESS,
+        c_2: DIMENSIONLESS,
+        l_g: MM,
+        l_kip: MM,
+        s: MM,
+    ) -> DIMENSIONLESS:
+        """Evaluates the formula, for more information see the __init__ method."""
+        raise_if_less_or_equal_to_zero(l_kip=l_kip)
+        raise_if_negative(c_1=c_1, l_g=l_g, s=s)
```
[the remark this answers is not in the document]
Comment (GerjanDorgelo) [reply blueprints/codes/eurocode/national_annex/nl/nen_en_1993_1_1_2006/chapter_NB_NB_critical_buckling_moment/formula_nb_nb_11.py]: continious would be 0

**Review by egarciamendez — DISMISSED** [review]
Checked against NEN-EN 1993-1-1+C2+A1:2016/NB:2016, NB.NB.4.1 to NB.NB.4.3, pages 57 to 60.
NB.NB.2, NB.NB.7, NB.NB.8 and NB.NB.12 match the printed formulas. Two defects, one affecting results.

## 1. `formula_nb_nb_11.py:37-38` and `:64` - sign convention of `C_2` inverted, negative value rejected

Annex, page 60: positive for the **onderflens** (bottom flange), negative for the **bovenflens** (top flange).
The docstring states the reverse.

`C_2` enters `pi * C_2 * S / L_kip` linearly. With the test values, `c_2 = +0.45` gives `C = 16.104`,
`c_2 = -0.45` gives `C = 8.065`. `M_cr` scales linearly with `C`, so a top flange load entered per the
docstring roughly doubles `M_cr`. Unsafe side.

Line 64 rejects negative `C_2`, which is the value the annex prescribes for top flange loading, and for the
linear extrapolation up to `0,1 h` above it.

Fix: swap top and bottom flange in the docstring, remove `c_2` from `raise_if_negative`, add a test with
`c_2 = -0.45` giving `C = 8.06457624644`.

## 2. `formula_nb_nb_9.py:60` - `max(575, ...)` is not in the annex

Annex, page 58: `α = (h × t_f × 10¹²) / (t_w³ × b × L_g²) > 575`. That is a condition of application, not a
floor on the result. The clamp returns 575 where the expression gives, say, 200.

Figure NB.NB.1, page 59: `k_red` is flat at 1,00 from `α = 0` to 555, then descends. The flat part comes from
`min(..., 1)` in NB.NB.8, kink at `0,03 / 5,4e-5 = 555,56`. Flooring `α` at 575 caps `k_red` at 0,99895, so
1,00 becomes unreachable.

Inconsistent with the same file's handling of `α <= 5 000`, which raises rather than clamps.

Fix: return the expression. The existing test uses `α = 1310` and cannot distinguish the two readings; add a
case below 575.

Separately: for `α > 5 000` the annex requires verification of the compressed edge per 6.3.3. Worth naming
6.3.3 in the `ValueError` message.

## 3. `formula_nb_nb_9.py:64` and `formula_nb_nb_11.py` - `complete_with_units` dimensionally wrong

`t_w^3` and `L_g^2` are written without brackets, so after substitution the exponent binds to the unit:

```
8.600 \ mm^3 \cdot 180.000 \ mm \cdot 6000.000 \ mm^2
```

That reads as 8,6 mm³ instead of `(8,600 mm)³`. Same for `S^2` and `L_{kip}^2` in NB.NB.11. Plain `complete`
is unaffected.

Fix: `\left(t_w\right)^3`, `\left(L_g\right)^2`.

## 4. Translations

- "kiplengte" -> "tip length". Should be lateral-torsional buckling length.
- "gaffels" -> "supports". Should be fork supports. Also affects `L_g` in NB.NB.2, defined as the length
  between the fork supports.
- "kipsteun" -> "lateral support". Should be lateral-torsional restraint.

## 5. Minor

- `formula_nb_nb_2.py`: bare `C`, `E`, `G` as substitution keys. Unique today, silently breakable on any edit
  to the equation string. Anchor them as you did with `r"h "`.
- NB.NB.7 renders as `k_red = 1 = 1 = 1.000 \ -`.
- `source_document` says `NEN-EN 1993-1-1:2006+A1:2014+NB:2016`, the document checked is headed
  `NEN-EN 1993-1-1+C2+A1:2016/NB:2016`. Constant predates this PR. Same annex text in both editions?

File: blueprints/codes/eurocode/national_annex/nl/nen_en_1993_1_1_2006/chapter_NB_NB_critical_buckling_moment/formula_nb_nb_11.py [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+
+        Parameters
+        ----------
+        c_1 : DIMENSIONLESS
+            [$C_1$] Coefficient dependent on the nature of the loading [-].
+        c_2 : DIMENSIONLESS
+            [$C_2$] Coefficient dependent on the location of application of the load relative to the neutral axis.
+            C₂ = 0 if the load acts at the centroid of the cross-section.
+            C₂ must be entered in the formula for determining coefficient C with a positive sign if the load acts at the centroid of the top flange.
+            C₂ must be entered in this formula with a negative sign if the load acts at the centroid of the bottom flange [-].
```
Comment (egarciamendez) [inline blueprints/codes/eurocode/national_annex/nl/nen_en_1993_1_1_2006/chapter_NB_NB_critical_buckling_moment/formula_nb_nb_11.py]: Reversed. Annex p.60: positive for the **onderflens** (bottom flange), negative for the **bovenflens** (top flange).

`C_2` enters `pi * C_2 * S / L_kip` linearly. With the test values `+0.45` gives `C = 16.104` and `-0.45` gives `C = 8.065`. `M_cr` scales linearly with `C`, so a top flange load entered per this docstring roughly doubles it.

File: blueprints/codes/eurocode/national_annex/nl/nen_en_1993_1_1_2006/chapter_NB_NB_critical_buckling_moment/formula_nb_nb_11.py [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+    def _evaluate(
+        c_1: DIMENSIONLESS,
+        c_2: DIMENSIONLESS,
+        l_g: MM,
+        l_kip: MM,
+        s: MM,
+    ) -> DIMENSIONLESS:
+        """Evaluates the formula, for more information see the __init__ method."""
+        raise_if_less_or_equal_to_zero(l_kip=l_kip)
+        raise_if_negative(c_1=c_1, l_g=l_g, s=s, c_2=c_2)
```
Comment (egarciamendez) [inline blueprints/codes/eurocode/national_annex/nl/nen_en_1993_1_1_2006/chapter_NB_NB_critical_buckling_moment/formula_nb_nb_11.py]: `c_2` has to be able to go negative: the annex prescribes a negative value for top flange loading, and for the linear extrapolation up to `0,1 h` above it. Drop it here and add a test with `c_2 = -0.45`, giving `C = 8.06457624644`.

File: blueprints/codes/eurocode/national_annex/nl/nen_en_1993_1_1_2006/chapter_NB_NB_critical_buckling_moment/formula_nb_nb_9.py [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+        t_f: MM,
+        t_w: MM,
+        b: MM,
+        l_g: MM,
+    ) -> DIMENSIONLESS:
+        """Evaluates the formula, for more information see the __init__ method."""
+        raise_if_less_or_equal_to_zero(t_w=t_w, b=b, l_g=l_g)
+        raise_if_negative(h=h, t_f=t_f)
+
+        return max(575, (h * t_f * 1e12) / (t_w**3 * b * l_g**2))
```
Comment (egarciamendez) [inline blueprints/codes/eurocode/national_annex/nl/nen_en_1993_1_1_2006/chapter_NB_NB_critical_buckling_moment/formula_nb_nb_9.py:60]: Annex p.58 prints `> 575` as a condition of application, not a floor on the result.

Figure NB.NB.1 shows `k_red` flat at 1,00 up to `alpha` around 555. Flooring `alpha` at 575 caps `k_red` at 0,99895, so 1,00 becomes unreachable. It is also inconsistent with `alpha <= 5 000` in NB.NB.8, which raises rather than clamps.

Return the expression. The test at `alpha = 1310` cannot distinguish the two readings; a case below 575 can.

File: blueprints/codes/eurocode/national_annex/nl/nen_en_1993_1_1_2006/chapter_NB_NB_critical_buckling_moment/formula_nb_nb_9.py [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+
+    def latex(self, n: int = 3) -> LatexFormula:
+        """Returns LatexFormula object for formula NB.NB.9."""
+        _equation: str = r"\max\left(575, \frac{h \cdot t_f \cdot 10^{12}}{t_w^3 \cdot b \cdot L_g^2}\right)"
```
Comment (egarciamendez) [inline blueprints/codes/eurocode/national_annex/nl/nen_en_1993_1_1_2006/chapter_NB_NB_critical_buckling_moment/formula_nb_nb_9.py]: `t_w^3` and `L_g^2` without brackets: after substitution the exponent binds to the unit, so `complete_with_units` renders `8.600 \ mm^3` instead of `(8.600 mm)^3`. Use `\left(t_w\right)^3` and `\left(L_g\right)^2`. Plain `complete` is unaffected.

File: blueprints/codes/eurocode/national_annex/nl/nen_en_1993_1_1_2006/chapter_NB_NB_critical_buckling_moment/formula_nb_nb_11.py [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+        term2 = np.sqrt(1 + (np.pi**2 * s**2) / (l_kip**2) * (c_2**2 + 1))
+        term3 = (np.pi * c_2 * s) / l_kip
+
+        return term1 * (term2 + term3)
+
+    def latex(self, n: int = 3) -> LatexFormula:
+        """Returns LatexFormula object for formula NB.NB.11."""
+        _equation: str = (
+            r"\frac{\pi \cdot C_1 \cdot L_g}{L_{kip}} \cdot \left( \sqrt{1 + \frac{\pi^2 \cdot S^2}{L_{kip}^2} "
+            r"\cdot \left(C_2^2 + 1\right)} + \frac{\pi \cdot C_2 \cdot S}{L_{kip}} \right)"
```
Comment (egarciamendez) [inline blueprints/codes/eurocode/national_annex/nl/nen_en_1993_1_1_2006/chapter_NB_NB_critical_buckling_moment/formula_nb_nb_11.py:75]: Same bracket issue as NB.NB.9: `S^2` and `L_{kip}^2` render as `1200.000 \ mm^2` in `complete_with_units`.

File: blueprints/codes/eurocode/national_annex/nl/nen_en_1993_1_1_2006/chapter_NB_NB_critical_buckling_moment/formula_nb_nb_2.py [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+        return k_red * (c / l_g) * np.sqrt(e * i_z * g * i_t)
+
+    def latex(self, n: int = 3) -> LatexFormula:
+        """Returns LatexFormula object for formula NB.NB.2."""
+        _equation: str = r"k_{red} \cdot \frac{C}{L_g} \cdot \sqrt{E \cdot I_z \cdot G \cdot I_t}"
+        _numeric_equation: str = latex_replace_symbols(
+            _equation,
+            {
+                r"k_{red}": f"{self.k_red:.{n}f}",
+                r"C": f"{self.c:.{n}f}",
```
Comment (egarciamendez) [inline blueprints/codes/eurocode/national_annex/nl/nen_en_1993_1_1_2006/chapter_NB_NB_critical_buckling_moment/formula_nb_nb_2.py]: Bare `C`, `E` and `G` as substitution keys. Unique in this string today, silently breakable on any edit to the equation. Anchor them as you did with `r"h "` in NB.NB.9.

File: blueprints/codes/eurocode/national_annex/nl/nen_en_1993_1_1_2006/chapter_NB_NB_critical_buckling_moment/formula_nb_nb_2.py [hunk 8]
```diff
@@ -0,0 +1,113 @@
...
+        Parameters
+        ----------
+        k_red : DIMENSIONLESS
+            [$k_{red}$] Reduction factor dependent on the degree of deformation of the beam cross-section with regard to beam length;
+            for calculating [$k_{red}$] applies NB.NB.4.2 [-].
+        c : DIMENSIONLESS
+            [$C$] Coefficient dependent on beam length and load point location, and support point of the load;
+            for calculating [$C$] applies NB.NB.4.3 [-].
+        l_g : MM
+            [$L_g$] Length of the beam between supports [$mm$].
```
Comment (egarciamendez) [inline blueprints/codes/eurocode/national_annex/nl/nen_en_1993_1_1_2006/chapter_NB_NB_critical_buckling_moment/formula_nb_nb_2.py]: Annex p.57: "de lengte van de ligger tussen de **gaffels**". Fork supports specifically, not supports in general.

File: blueprints/codes/eurocode/national_annex/nl/nen_en_1993_1_1_2006/chapter_NB_NB_critical_buckling_moment/formula_nb_nb_11.py [hunk 9]
```diff
@@ -0,0 +1,108 @@
...
+            [$C_1$] Coefficient dependent on the nature of the loading [-].
+        c_2 : DIMENSIONLESS
+            [$C_2$] Coefficient dependent on the location of application of the load relative to the neutral axis.
+            C₂ = 0 if the load acts at the centroid of the cross-section.
+            C₂ must be entered in the formula for determining coefficient C with a positive sign if the load acts at the centroid of the top flange.
+            C₂ must be entered in this formula with a negative sign if the load acts at the centroid of the bottom flange [-].
+        l_g : MM
+            [$L_g$] Length of the beam between the supports [$mm$].
+        l_kip : MM
+            [$L_{kip}$] Replaced unsupported tip length between two supports, between one support and one lateral support,
```
Comment (egarciamendez) [inline blueprints/codes/eurocode/national_annex/nl/nen_en_1993_1_1_2006/chapter_NB_NB_critical_buckling_moment/formula_nb_nb_11.py]: "kiplengte" is the lateral-torsional buckling length, not "tip length". Same line: "gaffels" are fork supports and "kipsteunen" are lateral-torsional restraints.

**Review by egarciamendez — CHANGES_REQUESTED** [review]
Checked against NEN-EN 1993-1-1+C2+A1:2016/NB:2016, NB.NB.4.1 to NB.NB.4.3, pages 57 to 60. NB.NB.2, NB.NB.7, NB.NB.8 and NB.NB.12 match the printed formulas.

File: blueprints/codes/eurocode/national_annex/nl/nen_en_1993_1_1_2006/chapter_NB_NB_critical_buckling_moment/formula_nb_nb_9.py (hunk 4 above)
Comment (GerjanDorgelo) [reply blueprints/codes/eurocode/national_annex/nl/nen_en_1993_1_1_2006/chapter_NB_NB_critical_buckling_moment/formula_nb_nb_9.py:60]: im not sure what to do here.

**Review by egarciamendez — APPROVED** [review]
🚀
